blocks/eda-frontend/src/components/Shared/Graph.js

```diff
@@ -31,10 +31,21 @@ class Graph extends React.Component {
               const chart = this
               const series = this.series[0]
               const starttime = Date.now()
+
               function addPoints () {
                 while (!pointList.isEmpty()) {
+                  const xmax = datapoint.datapointXMax
                   const point = pointList.peek()
                   const x = parseFloat(point[1])
+                  if (x > xmax) {
+                    datapoint.datapointXMax = datapoint.datapointXMax + 4
+                    datapoint.datapointXMin = datapoint.datapointXMin + 4
```

Comment:
```suggestion
                    const xmin = chart.xAxis[0].min
                    const xshift = (xmax - xmin) * 0.2
```
Don't use datapoint. Directly get from chart.

Also, calculate the shift from the difference. Using 4 may not always work.